artifactory/cli.go

```diff
@@ -1938,6 +1938,9 @@ func uploadCmd(c *cli.Context) error {
 	if c.NArg() > 0 && c.IsSet("spec") {
 		return cliutils.PrintHelpAndReturnError("No arguments should be sent when the spec option is used.", c)
 	}
+	if c.IsSet("regexp") && c.IsSet("ant") {
```

Comment:
This check should be added into the ```spec.ValidateSpec``` function, Notice that this function is called a few lines below. Letting this function handle the validation, takes care fo both the flags and the file-spec validation.
Something else to notice:
The following flag combinations are valid:
```
--ant --regexp=false
--ant=false --regexp
--ant=treu --regexp=fase
```
and so on.
The ```spec.ValidateSpec``` function takes care of all of that very easilly.